homeassistant/components/fritzbox/__init__.py

```diff
@@ -11,24 +11,27 @@
     CONF_USERNAME,
     EVENT_HOMEASSISTANT_STOP,
 )
-from homeassistant.helpers import discovery
 import homeassistant.helpers.config_validation as cv
 
-_LOGGER = logging.getLogger(__name__)
-
-SUPPORTED_DOMAINS = ["binary_sensor", "climate", "switch", "sensor"]
+from .const import (
+    CONF_CONNECTIONS,
+    DEFAULT_HOST,
+    DEFAULT_USERNAME,
+    DOMAIN,
+    SUPPORTED_DOMAINS,
+)
 
-DOMAIN = "fritzbox"
 
-ATTR_STATE_BATTERY_LOW = "battery_low"
-ATTR_STATE_DEVICE_LOCKED = "device_locked"
-ATTR_STATE_HOLIDAY_MODE = "holiday_mode"
-ATTR_STATE_LOCKED = "locked"
-ATTR_STATE_SUMMER_MODE = "summer_mode"
-ATTR_STATE_WINDOW_OPEN = "window_open"
+def ensure_unique_hosts(value):
+    """Validate that all configs have a unique host."""
+    vol.Schema(vol.Unique("duplicate host entries found"))(
+        [socket.gethostbyname(entry[CONF_HOST]) for entry in value]
+    )
+    return value
 
 
 CONFIG_SCHEMA = vol.Schema(
+    cv.deprecated(DOMAIN),
```

Comment:
We're missing a `vol.All` here so that both the deprecation validator and the rest of the schema validation are used.